[18.0][MIG] account_invoice_section_sale_order: Migration to 18.0

LucasTran380381 opened the pull request:
Changes in v18:

* Due to changes in Odoo v18, `sale_order_line` `name` now include product names. This PR updates test cases to reflect this change.

antespi committed:
account_refund_original

Enhancements:

* Reverse link from invoice to refunds also available.
* Link at invoice line level, not only at invoice level.
* When installing the module after some refunds have been made, the module tries to find the link to the original invoice for them.
* OCA Guidelines compliance.
* Compatibility in OpenUpgrade for migrating from 8.0 module account_refund_original.
* Tests with 100% coverage.

Technical changes:

* Rename field origin_invoices_ids to origin_invoice_ids
* Rename field refund_invoices_description to refund_reason
* Rename addon to account_invoice_refund_link

sbidoul committed:
remove en.po that was erroneously created by transbot

pedrobaeza committed:
Make modules uninstallable

pedrobaeza committed:
Rename manifest files

pedrobaeza committed:
account_invoice_refund_link: Improvements + cleaning

Odoo already includes a method for preparing the refund, so using it
we don't have to make any guess about the returned domain and the
method also serves for refunds created directly by code.

Initial line match has also been improved comparing product or
description.

luismontalba committed:
account_invoice_refund_link: Migrated to 10.0

Commit:
fix init hook by matching invoices and refunds only if in same company

oihane committed:
account_invoice_refund_link: Migration to 11.0

pedrobaeza committed:
account_invoice_refund_link: Don't copy m2m fields (#339)

* [FIX] account_invoice_refund_link: Don't copy m2m fields

Many2many fields have copy=True property by default, having
an incorrect duplication of origin or refund invoices.

Steps to reproduce:

* Create an invoice
* Refund it
* Duplicate the original invoice
* The refund will be link to 2 original invoices

Included tests covering the use cases for avoiding future regressions.

pedrobaeza committed:
account_invoice_refund_link: Don't choke with standard fields

Since v11, Odoo counts with a field called `refund_invoice_ids` of type
one2many, which conflicts with the m2m field defined in this module.
Moreover, these fields are not needed anymore.

There is only 2 OCA modules using this module right now, and they
don't need significant changes for this adaptation, but I will do it
when this get merged.

pedrobaeza committed:
account_invoice_refund_link: Pylint

Nikul-Chaudhary committed:
[MIG] account_invoice_refund_link v11 to v12

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (17 of 17 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_refund_link/de/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (17 of 17 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_refund_link/pt_BR/

ernestotejeda committed:
[IMP] account_invoice_refund_link: black, isort

ernestotejeda committed:
[MIG] account_invoice_refund_link: Migration to 13.0

TDu committed:
[13.0][ADD] account_invoice_section_sale_order

Description from the readme :

When invoicing multiple sale orders at the same time, sale orders may be grouped
by customer into a single invoice. Unfortunately when this happens, it is hard
to know which invoice line belongs to which sale order.

This module helps by grouping invoicing lines into sections with the name of the
targeted sale order.
This is only done when an invoice targets multiple sale order.

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (11 of 11 strings)

Translation: account-invoicing-13.0/account-invoicing-13.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-13-0/account-invoicing-13-0-account_invoice_refund_link/pt_BR/

sergio-teruel committed:
[FIX] account_invoice_refund_link: Fix foreignkey exception when a credit note invoice has done from sale order.

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-13.0/account-invoicing-13.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-13-0/account-invoicing-13-0-account_invoice_refund_link/pt_BR/

pedrobaeza committed:
[OU] account_invoice_refund_link: Migration scripts

phuctranfxvn committed:
[IMP] account_invoice_section_sale_order: black, isort, prettier

phuctranfxvn committed:
[MIG] account_invoice_section_sale_order: Migration to 14.0

joao-p-marques committed:
[MIG] account_invoice_refund_link: Migration to 14.0

sebalix committed:
[IMP] add a hook to get ordered invoice lines

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-14.0/account-invoicing-14.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-14-0/account-invoicing-14-0-account_invoice_refund_link/es/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 83.3% (10 of 12 strings)

Translation: account-invoicing-14.0/account-invoicing-14.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-14-0/account-invoicing-14-0-account_invoice_refund_link/fr/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-14.0/account-invoicing-14.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-14-0/account-invoicing-14-0-account_invoice_refund_link/es_AR/

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 91.6% (11 of 12 strings)

Translation: account-invoicing-14.0/account-invoicing-14.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-14-0/account-invoicing-14-0-account_invoice_refund_link/fr_FR/

pedrocs-exo committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-14.0/account-invoicing-14.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-14-0/account-invoicing-14-0-account_invoice_refund_link/pt_PT/

acsonefho committed:
[14.0][IMP] acount_invoice_refund_link: add index

OriolVForgeFlow committed:
[15.0][MIG] account_invoice_tree_currency: Migration to 15.0

grindtildeath committed:
account_invoice_section_sale_order: Custom section name

Refactor preparation of sections

Refactor to allow different section grouping

Add field to allow different possibilities of grouping invoice lines from sale order

TDu committed:
account_invoice_section_sale_order fix create_invoice

The function signature as changed in v14

koenloodts committed:
[MIG] account_invoice_refund_link: Migration to 15.0

ValentinVinagre committed:
[FIX] account_invoice_refund_link: Fix development_status

Commit:
[IMP] account_invoice_section_sale_order: black, isort, prettier

Commit:
[mig] Odoo 15 fixes

Commit:
[15.0] fix remove unused date-parameter from _create_invoices

sebastienbeau committed:
[FIX] fix passing default_journal_id when invoicing

sebastienbeau committed:
[FIX] account_invoice_section_sale: fix invoice total when using currency

pablontura committed:
Translated using Weblate (Catalan)

Currently translated at 91.6% (11 of 12 strings)

Translation: account-invoicing-15.0/account-invoicing-15.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-15-0/account-invoicing-15-0-account_invoice_refund_link/ca/

pedrobaeza committed:
[FIX] account_invoice_refund_link: Don't link incorrect lines

With the previous algorithm, if something modifies the order of the
returned lines, there are chances of linking invoice lines with non
refund lines (but other journal items).

We assure at least that only refund lines are linked restricting the
iteration on them.

TT38633

ljsalvatierra-factorlibre committed:
[MIG] account_invoice_refund_link: Migration to 16.0

CRogos committed:
[15.0][FIX] Remove tests from init

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_refund_link/hr/

pedrobaeza committed:
[FIX] account_invoice_section_sale_order: Avoid problems in integration tests

In OCA repo, now we have this error:

 Traceback (most recent call last):
  File "/__w/account-invoicing/account-invoicing/account_invoice_section_sale_order/tests/test_invoice_group_by_sale_order.py", line 109, in test_create_invoice
    self.assertEqual(line.name, result[line.sequence][0])
KeyError: 1

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_refund_link/pt/

Commit:
[FIX] account_invoice_refund_link: Exclude section/notes origin lines

On the side by side comparison, exclude the lines that are not products, as that's what we do on the refund lines.

rven committed:
[MIG] account_invoice_section_sale_order: Migration to 16.0

[UPD] Update account_invoice_section_sale_order.pot

[UPD] README.rst

andrel-exo committed:
[16.0][IMP] added tree with refunded lines

pedrobaeza committed:
[IMP] account_invoice_refund_link: Make more reliable the linking

Hooking on a high level method like `_reverse_moves` (although still
being private), makes that other modules hooking into it may alter the
number of returned lines (like for example,
account_invoice_refund_line_selection).

We choose the low-level `copy_data` method that is used in such method
for linking the refund lines with their origin ones, avoiding the
problem.

Gelo-fl committed:
Added translation using Weblate (Spanish)

Translated using Weblate (Spanish)

Currently translated at 100.0% (1 of 1 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_section_sale_order
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_section_sale_order/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_refund_link/es/

josep-tecnativa committed:
[IMP] account_invoice_refund_link: test performance improvement
- Include context keys for avoiding mail operations overhead.

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_refund_link/pt_BR/

NICO-SOLUTIONS committed:
[IMP] account_invoice_section_sale_order: pre-commit auto fixes

NICO-SOLUTIONS committed:
[MIG] account_invoice_section_sale_order: Migration to 17.0

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-16.0/account-invoicing-16.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-16-0/account-invoicing-16-0-account_invoice_refund_link/it/

andrel-exo committed:
[IMP] account_invoice_refund_link: pre-commit auto fixes

andrel-exo committed:
[MIG] account_invoice_refund_link: Migration to 17.0

Commit:
[UPD] Update account_invoice_refund_link.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_refund_link/

manuelregidor committed:
[FIX] account_invoice_refund_link: Form view for refund invoices

Commit:
[UPD] Update account_invoice_refund_link.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_refund_link/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_refund_link/sv/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_refund_link/it/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (12 of 12 strings)

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_refund_link/pt/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_section_sale_order
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_section_sale_order/it/

sergiocorato committed:
[FIX] account_invoice_section_sale_order check line_ids exists

mmequignon committed:
account_invoice_section_sale_order: Fix read/write access

the `sale.order._create_invoices() method does not only requires
`create` access to the `account.move` and `account.move.line` models.

Because we're actually sorting lines after the creating, it requires
`read` access to sort, and `write` access to modify their sequence.

This is problematic when interracting with other modules like
`invoice_mode_at_shipping` where stock users are the ones to create
invoices.

This commit adds a few `sudo()` in the code, in order to avoid granting
`read` and `write` access to users that shouldn't be allowed to read and write invoices.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (18 of 18 strings)

Translation: account-invoicing-17.0/account-invoicing-17.0-account_invoice_section_sale_order
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-17-0/account-invoicing-17-0-account_invoice_section_sale_order/it/

Commit:
[IMP] account_invoice_section_sale_order: pre-commit auto fixes

Commit:
[MIG] account_invoice_section_sale_order: Migration to 18.0

thienvh332 reviewed (approved):
LGTM

thienvh332 commented:
Hi @LucasTran380381 ,
> Due to changes in Odoo v18, sale_order_line name now include product names. This PR updates test cases to reflect this change.

==> It's not `sale_order_line` but `account_move_line`

This change is due to https://github.com/odoo/odoo/commit/b647d202efafd96f71312113484a3552dc89dc05#diff-78f3e1847de8ca0acf28d72a412947a549acdb08142d1dadf7646363d7972cb0R518. When creating PR on OCA, please add it to the description.

Commit:
[MIG] account_invoice_refund_link: Migration to 18.0

OCA-git-bot committed:
Merge PR #1850 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update account_invoice_refund_link.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-18.0/account-invoicing-18.0-account_invoice_refund_link
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-18-0/account-invoicing-18-0-account_invoice_refund_link/

simahawk committed:
account_invoice_section_sale_order: fix access right tests not running

simahawk committed:
account_invoice_section_sale_order: fix access right on move line write
